ui/widgets/datepicker.js

```diff
@@ -79,30 +81,33 @@ var widget = $.widget( "ui.datepicker", {
 	_getCreateOptions: function() {
 		var max = this.element.attr( "max" ),
 			min = this.element.attr( "min" ),
+			parser = new Globalize( "en" ).dateParser( { raw: "yyyy-MM-dd" } ),
```

Comment:
No idea why, but I was focused on the missing hour not the month. Stupid me.

Pushed a new commit with this.